python/delta-kernel-rust-sharing-wrapper/src/lib.rs

```diff
@@ -78,6 +80,38 @@ impl ScanBuilder {
     }
 }
 
+fn try_get_schema(schema: &Arc<StructType>) -> Result<SchemaRef, delta_kernel::Error> {
+    Ok(Arc::new(schema.as_ref().try_into().map_err(|e| {
+        delta_kernel::Error::Generic(format!("Could not get result schema: {e}"))
+    })?))
+}
+
+// TODO(patrick.jin): Change return type to RecordBatchIterator<impl Iterator<...>>
```

Comment:
should we finish this TODO before merging?

Reply:
I tried to but it seems non-trivial. But I will try to address it before the final release.

More specifically, changing the Vec to an Iterator requires the return lifetime of `Scan::execute` and `TableChangesScan::execute` to be `'_`, and that requires the return type of those functions to be `...<Arc<...>>` insteadof `...<Box<...>>`. And upon making that change, I get a `cannot be converted to a python object` error. Will see if @zachschuermann or @OussamaSaoudi-db knows more about this error.

Reply:
Posted a fix this morning, so this is now fixed :)